api/v1beta1/types_class.go

```diff
@@ -52,6 +56,15 @@ type AzureClusterClassSpec struct {
 	CloudProviderConfigOverrides *CloudProviderConfigOverrides `json:"cloudProviderConfigOverrides,omitempty"`
 }
 
+// ExtendedLocationSpec defines the ExtendedLocation properties to enable CAPZ for Azure public MEC.
+type ExtendedLocationSpec struct {
+	// Name defines a name for extended location.
+	Name string `json:"name"`
+
+	// Type defines a type for extended location (eg. EdgeZone)
```

Comment:
```suggestion
	// Type defines a type for extended location (eg. EdgeZone).
```

Reply:
is there a well known set of valid type values? If so we should validate this with a kubebuilder Enum

Reply:
Yes, I have added the Enum for type value.